src/tree_cmd.cpp

```diff
@@ -80,6 +81,34 @@ static bool CanPlantTreesOnTile(TileIndex tile, bool allow_desert)
 	}
 }
 
+/**
+* Tests if a tile is near an already existing forest.
+* Trees won't spread if not placed near some other trees.
+*
+* @param tile the tile of interest
+* @return true if the tile is near a forest
+*/
+static bool IsNearbyForest(TileIndex tile)
+{
+	uint planted_tile_count = 0;
+
+	// An already planted tilte can always be planted again
+	if (IsTileType(tile, MP_TREES)) return true;
+
+	// Count the trees arround the clear tile to determine if it's near a forest
+	for (int x = -2; x <= 2; x++) {
```

Comment:
You might use `CircularTileSearch()` for this?

Reply:
it looks like that only returns a boolean rather than an integer, unfortunately.

Reply:
`if CircularTileSearch(IsTreeCallback) planted_tile_count++;` ?

Reply:
wouldn't that hit the first tree every time?

Reply:
Oh, er, yes. I think you can pass a reference to CicularTileSearch that you can update. I'd suggest looking at other uses to see if anything is applicable.

Reply:
Set up a `TileArea` and iterate with that?

Also tiles can have 1 to 3 trees on them, not sure if this should be taken into account.